### PR TITLE
spack-enable.bash: add option to set config directory via env variable

### PR DESCRIPTION
**Pull request by harshula**
Closes https://github.com/ACCESS-NRI/spack-config/issues/30

### CODE REVIEW

**Review by aidanheerdegen — APPROVED** [review]
Just to be clear, CI/CD would use the same local `spack` repo, but clone this `config` repo as required to a temporary directory, and choose a `spack-config` version to use for the CI/CD process via a command line argument?

**Comment by harshula** [comment]
> clone this `config` repo as required to a temporary directory

The spack-config directory must be in the same parent directory as `spack`. I need to make a change to allow the name to be different.

File: spack-enable.bash [hunk 1]
```diff
@@ -18,15 +18,53 @@
...
+    fi
+
+    echo "Using configuration in ${CONFIG}"
+    export SPACK_DISABLE_LOCAL_CONFIG="false"
+    export SPACK_USER_CONFIG_PATH="${CONFIG}"
+    export SPACK_SYSTEM_CONFIG_PATH="${CONFIG}"
+    unset CONFIG
+
+else
+    echo "Usage: . ${PROGNAME} [<spack-version>/{ci,gadi}]"
```
Comment (aidanheerdegen) [inline spack-enable.bash:64]: I don't understand this intended usage. Is it expected to be able to invoke it like so:
```
. spack-enable.bash 0.22/ci
```
for non-CI/CD?

Or
```
. spack-enable.bash 0.22/gadi
```
for CI/CD?

I'm thinking of the intention is for a version to be specified only for CI/CD then
```
. spack-enable.bash 0.22
```
would be sufficient, and you could add `ci` to the `0.22` path automatically. 

Or change the help to
```
echo "Usage: . ${PROGNAME} [<spack-version>/ci]"
```

Am I understanding the intention of this correctly?

**Review by aidanheerdegen — APPROVED** [review]
Approving in case my question is nonsense and you just want to merge.

File: spack-enable.bash (hunk 1 above)
Comment (harshula) [reply spack-enable.bash:64]: For CI: `. spack-enable.bash 0.22/ci`
For CD: `. spack-enable.bash 0.22/gadi`

**Review by aidanheerdegen — APPROVED** [review]
LGTM

**Pull request merged**